### DISCUSSION TITLE
Operation not supported by the backend (-2700)

### DISCUSSION THREAD

**Discussion by elpsk** [utmapp/UTM#5223] [macOS] [answered]
In the latest Pre-Release (4.2.3 beta: [https://github.com/utmapp/UTM/releases/tag/v4.2.3](https://github.com/utmapp/UTM/releases/tag/v4.2.3)) I read:

> (macOS) Guest interface commands. New commands accessible from the scripting interface as well as the command line interface (utmctl) can be used to read/write files, execute commands, and list IP addresses. **These commands require QEMU guest agent to be installed.**

What is the procedure to install "**QEMU guest agent**" on macOS?

I've used brew: `$ brew install qemu` but seems not enough. 

When I use this command to launch a cmd command on a macOS VM, I receive this error:

```
$ ./utmctl exec XXXXX-XXX-XXX --cmd "/bin/sh ~/Desktop/execute.sh"
Error from event: The operation couldn’t be completed. (OSStatus error -2700.)
Operation not supported by the backend.
$ 
```

thanks

**Comment by osy** [comment, marked as answer]
It’s not supported on macos guests.

**Reply by elpsk** [reply]
There is another way to use commands?
My idea is to use UTM in a pipeline. Launching commands became very useful.

**Reply by diesieben07** [reply]
Agreed, it would be nice to at least have a way to get the IP address of the VM. Then I could use SSH to do anything else.

**Reply by naikrovek** [reply]
What about Linux guests?  I get this error on a Linux guest when trying to run `utmctl ip-address` and the name of my VM.
